### PR TITLE
MVP on Calendar component

### PR DESCRIPTION
**Pull request by IamTechknow**

### CODE REVIEW

File: client/components/Calendar.jsx [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    // Lookup tables to quickly get information
+    this.days = ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa'];
+    this.daysInMonth = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+    this.monthName = ['January', 'Feburary', 'March', 'April', 'May', 'June',
+      'July', 'August', 'September', 'October', 'November', 'December'];
+  }
+
+  // Access reserved API and return a set with all reserved data in the month in a Promise.
+  loadReserved(id, month, year) {
+    return fetch(`${server}/api/listings/${id}/reserved?month=${month}&year=${year}`)
```
Comment (VictorOfCode) [inline client/components/Calendar.jsx:55]: Nice use of template literals

File: client/components/Calendar.jsx [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+  // If neither check day has been set, set the check in date
+  // If checkIn has been set, set the checkout date
+  onCellClick(month, year, event) {
+    // Validate the date is correct, first by checking the event target text
+    const day = Number.parseInt(event.target.textContent);
+
+    if (day && !this.state.reservedSet.has(day)) {
+      const clickDate = new Date(year, month, day);
+      const midnightTmw = new Date().setHours(24, 0, 0, 0);
```
Comment (VictorOfCode) [inline client/components/Calendar.jsx:117]: Perhaps a cleaner variable name is necessary?

File: test/__tests__/CalendarTest.jsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  });
+  
+  it('should display the correct number of rows based on the month', function() {
+    // TODO: Doesn't take props yet, this will need to be refactored too.
+    let calendar = mount(
+      <Calendar small />
+    );
+    
+    // Locate the right sided button by selecting with two classes, simulate clicks
+    var rightBtn = calendar.find('.btn-sm.checkoutFloatRight');
```
Comment (VictorOfCode) [inline test/__tests__/CalendarTest.jsx]: Perhaps you can reassign with "let" or "const" to stay consistent with airbnb style guide

File: test/__tests__/CalendarTest.jsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    
+    // Find the number of calendar rows, verify the correct number based on the current month
+    var rows = calendar.find('.checkoutCalRow');
```
Comment (VictorOfCode) [inline test/__tests__/CalendarTest.jsx]: "let" or "const"

File: test/__tests__/CalendarTest.jsx (hunk 3 above)
Comment (VictorOfCode) [inline test/__tests__/CalendarTest.jsx]: "let" or "const"

Comment (IamTechknow) [reply test/__tests__/CalendarTest.jsx]: Fixed

File: test/__tests__/CalendarTest.jsx (hunk 4 above)
Comment (IamTechknow) [reply test/__tests__/CalendarTest.jsx]: Fixed

File: test/__tests__/CalendarTest.jsx (hunk 3 above)
Comment (IamTechknow) [reply test/__tests__/CalendarTest.jsx]: Fixed

**Pull request merged**